registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/logger/RegProcessorLogger.java

```diff
@@ -24,6 +31,20 @@ private RegProcessorLogger() {
 	 * @return the logger
 	 */
 	public static Logger getLogger(Class<?> clazz) {
-		return Logfactory.getSlf4jLogger(clazz);
+		Map<String,String> map = new HashMap<String, String>();
+		Logger logger=  Logfactory.getSlf4jLogger(clazz);
+		if(map.isEmpty()) {
+			Properties properties=System.getProperties();
+			LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+			properties.forEach((key, value) -> {
+			      if (key.toString().startsWith(PROP_PREFIX)) {
+			        map.put(key.toString().substring(PROP_PREFIX.length() + 1), value.toString());
+			      }
+			    });
+				map.forEach((key, value) -> {
+					loggerContext.getLogger(key).setLevel(Level.valueOf(value));

```

Comment:
Let's maintain a static class-level map to store the filtered logging level config key-values. - will be executed only once if the map is empty.
Also, let's have a null check on loggerContext.getLogger(key) method before setting the log level. and the log level should be set on the current logger instance.

Reply:
Fixed code as per comment